plugins/arElasticSearchPlugin/lib/arElasticSearchPlugin.class.php

```diff
@@ -479,7 +507,11 @@ protected function initialize()
                     && isset($this->config['index']['configuration']['analysis']['char_filter']['strip_md'])
                 ) {
                     foreach ($this->config['index']['configuration']['analysis']['analyzer'] as $key => $analyzer) {
-                        $this->config['index']['configuration']['analysis']['analyzer'][$key]['char_filter'] = ['strip_md'];
+                        if ((in_array($key, ['english', 'french', 'default', 'autocomplete'])) && sfConfig::get('app_diacritics')) {
+                            $this->config['index']['configuration']['analysis']['analyzer'][$key]['char_filter'] = ['strip_md', 'diacritics_lowercase'];
+                        } else {
+                            $this->config['index']['configuration']['analysis']['analyzer'][$key]['char_filter'] = ['strip_md'];
+                        }
```

Comment:
Why only English and French? I think it will be more useful for other cultures if the language settings are properly set. Also, as we discussed a while ago, we should improve the previous implementation and not replace the char filter entirely, but merge it with the existing one if one is set; it won't be an issue right now but it could be one in the future if someone adds a char filer in the configuration.

Reply:
i've updated code to add `diacritics_lowercase` to all analyzers now!

Reply:
Thanks @melaniekung! But it's still replacing the existing char filters set in the configuration (`$this->config['index']['configuration']['analysis']['analyzer'][$key]['char_filter']`).

Right now we don't have any in the default config so it won't be an issue, but they may be added in the future and it will be hard to troubleshot why are they not being applied, as it happened to you when you started this work.

Reply:
how about replacing it after checking for current values ??

Reply:
I'd check if it's set and merge the arrays if it's. This is what I shared with you at that time, the current implementation has changed a bit but it should be similar:

```php
foreach ($this->config['index']['configuration']['analysis']['analyzer'] as $key => $analyzer) {
    $filters = ['strip_md'];
    if ($this->config['index']['configuration']['analysis']['analyzer'][$key]['char_filter']) {
        $filters = array_merge($filters, $this->config['index']['configuration']['analysis']['analyzer'][$key]['char_filter']);
    }
    $this->config['index']['configuration']['analysis']['analyzer'][$key]['char_filter'] = $filters;
}
```

Reply:
sorry , i thought i already implemented that :sweat_smile:

Reply:
Thanks!